include/xgboost/learner.h

```diff
@@ -9,13 +9,13 @@
 #define XGBOOST_LEARNER_H_
 
 #include <rabit/rabit.h>
-
 #include <xgboost/base.h>
 #include <xgboost/gbm.h>
 #include <xgboost/metric.h>
 #include <xgboost/objective.h>
 #include <xgboost/feature_map.h>
 #include <xgboost/generic_parameters.h>
+#include <xgboost/json.h>
```

Comment:
Can we substitute this with a forward declaration
class Json;
?

It seems Json is only used as a pointer and reference in this header

Reply:
@trams I moved the `ModelParameter` into this header.  Maybe I should move it into an independent header.

Reply:
I see.
It is not very obvious that json.h has something besides json handling